lib/ide/ast/impl/src/lib.rs

```diff
@@ -152,6 +152,28 @@ impl Ast {
         self
     }
 
+    // TODO use strongly typed IDMap, after it's merged to master
+
+    fn collect_ids(&self, index:&mut usize, ids:&mut IDMap) {
+        self.id.map( |id| {
```

Comment:
There are two issues:
1. We count span twice  — once for `self`, once for its children. But self's span already includes the children.
2. We can't count index by adding subsequent children's spans, as they are not continuous.
Eg. `a + b` must also take into account spacing around the operator and it is not included in any of the infix application child.

I think we need to start with implementing a new algorithm like `zipWithOffset`.

Reply:
+ Please fix linter warnings - run `cargo clippy` localy to get them :)